--- flite-2.2/config/common_make_rules.orig	2020-08-13 02:17:09.000000000 +0200
+++ flite-2.2/config/common_make_rules	2020-09-18 18:53:43.009784438 +0200
@@ -68,8 +68,22 @@
 
 FLITELIBS = $(BUILDDIR)/lib/libflite.a
 FLITELIBFLAGS = -L$(BUILDDIR)/lib -lflite 
-LDFLAGS += -lm $(AUDIOLIBS) $(OTHERLIBS)
+LDFLAGS += -lm $(OTHERLIBS)
 
+libflite_LDLIBS = $(AUDIOLIBS) -lm
+libflite_cmulex_LDLIBS = -lflite
+libflite_usenglish_LDLIBS = -lflite
+libflite_cmu_grapheme_lang_LDLIBS = -lflite
+libflite_cmu_grapheme_lex_LDLIBS = -lflite
+libflite_cmu_indic_lang_LDLIBS = -lflite_cmulex -lflite_usenglish -lflite
+libflite_cmu_indic_lex_LDLIBS = -lflite_cmu_indic_lang -lflite_cmulex -lflite_usenglish -lflite
+libflite_cmu_us_awb_LDLIBS = -lflite_cmulex -lflite_usenglish -lflite
+libflite_cmu_us_kal_LDLIBS = -lflite_cmulex -lflite_usenglish -lflite
+libflite_cmu_us_kal16_LDLIBS = -lflite_cmulex -lflite_usenglish -lflite
+libflite_cmu_us_rms_LDLIBS = -lflite_cmulex -lflite_usenglish -lflite
+libflite_cmu_us_slt_LDLIBS = -lflite_cmulex -lflite_usenglish -lflite
+libflite_cmu_time_awb_LDLIBS = -lflite -lflite_cmulex -lflite_usenglish -lflite
+ 
 FULLOBJS = $(OBJS:%=$(OBJDIR)/%)
 ifdef SHFLAGS
 SOOBJS = $(OBJS:.o=.os)
@@ -123,7 +137,7 @@
 	@ rm -rf shared_os && mkdir shared_os
 	@ rm -f $@ $@.${PROJECT_VERSION} $@.${PROJECT_SHLIB_VERSION} 
 	@ (cd shared_os && ar x ../$<)
-	@ (cd shared_os && $(CC) -shared -Wl,-soname,`basename $@`.${PROJECT_SHLIB_VERSION} -o ../$@.${PROJECT_VERSION} *.os $(LDFLAGS))
+	(cd shared_os && $(CC) -shared -Wl,-soname,`basename $@`.${PROJECT_SHLIB_VERSION} -Wl,-rpath-link=../$(dir $@) -L../$(dir $@) -o ../$@.${PROJECT_VERSION} *.os $(LDFLAGS)  $($(patsubst %.so,%_LDLIBS,$(notdir $@))))
 	@ (cd $(LIBDIR) && ln -s `basename $@.${PROJECT_VERSION}` `basename $@.${PROJECT_SHLIB_VERSION}` )
 	@ (cd $(LIBDIR) && ln -s `basename $@.${PROJECT_SHLIB_VERSION}` `basename $@` )
 	@ rm -rf shared_os
--- flite-1.2-release/doc/Makefile..mk	2002-12-30 15:57:30 +0300
+++ flite-1.2-release/doc/Makefile	2006-04-10 01:12:41 +0400
@@ -40,7 +40,7 @@
 ALL_DIRS=
 FILES = Makefile flite.texi intro.txt alice stuff.ed
 
-ALL = 
+ALL = flite.html
 
 include $(TOP)/config/common_make_rules
 
--- flite-2.2/main/Makefile.orig	2020-08-13 02:17:09.000000000 +0200
+++ flite-2.2/main/Makefile	2020-09-18 19:06:56.732151141 +0200
@@ -54,10 +54,10 @@
 VOICES=$(VOXES)
 VOICELIBS=$(VOICES:%=flite_%)
 
-flite_MODS = $(VOICELIBS) $(LANGS:%=flite_%) $(LEXES:%=flite_%) 
-flite_LIBS = $(flite_MODS) flite
+flite_MODS = $(LEXES:%=flite_%) $(LANGS:%=flite_%)  $(VOICELIBS)
+flite_LIBS = flite $(flite_MODS)
 
-flite_LIBS_flags = -L$(LIBDIR) $(flite_LIBS:%=-l%)
+flite_LIBS_flags = -L$(LIBDIR) -Wl,-rpath-link=$(LIBDIR) $(flite_LIBS:%=-l%)
 flite_LIBS_deps = $(flite_LIBS:%=$(LIBDIR)/lib%.a)
 
 include $(TOP)/config/common_make_rules
@@ -74,7 +74,7 @@
                     $(SHAREDLIBS:%=%.${PROJECT_SHLIB_VERSION})
 
 flite_time_LIBS = flite_cmu_time_awb flite_usenglish flite_cmulex
-flite_time_LIBS_flags = -L$(LIBDIR) $(flite_time_LIBS:%=-l%)
+flite_time_LIBS_flags = -L$(LIBDIR) -Wl,-rpath-link=$(LIBDIR) $(flite_time_LIBS:%=-l%)
 flite_time_LIBS_deps = $(flite_time_LIBS:%=$(LIBDIR)/lib%.a)
 
 LOCAL_CLEAN = $(BINDIR)/flite$(EXEEXT) $(BINDIR)/flite_time$(EXEEXT) \
--- flite-2.1-release/Makefile.orig	2017-12-25 16:13:21.000000000 +0100
+++ flite-2.1-release/Makefile	2019-04-20 09:16:48.529713034 +0200
@@ -46,7 +46,7 @@
 BUILD_DIRS = include src lang doc
 ALL_DIRS=config $(BUILD_DIRS) testsuite \
          wince windows android \
-         sapi tools main 
+         sapi main tools
 CONFIG=configure configure.in config.sub config.guess \
        missing install-sh mkinstalldirs
 OLD_WINDOWS = Exports.def flite.sln fliteDll.vcproj
@@ -56,7 +56,7 @@
                 config.cache config.log config.status \
 		config/config config/system.mak FileList
 
-HOST_ONLY_DIRS = tools main
+HOST_ONLY_DIRS = main tools
 ALL = $(BUILD_DIRS)
 
 config_dummy := $(shell test -f config/config || ( echo '*** '; echo '*** Making default config file ***'; echo '*** '; ./configure; )  >&2)
