--- flite-1.4-release/Makefile.orig	2012-10-06 12:50:46.580667756 +0200
+++ flite-1.4-release/Makefile	2012-10-06 13:54:32.223921351 +0200
@@ -106,10 +106,10 @@
 
 install:
 	@echo Installing 
-	mkdir -p $(INSTALLBINDIR)
-	mkdir -p $(INSTALLLIBDIR)
-	mkdir -p $(INSTALLINCDIR)
-	$(INSTALL) -m 644 include/*.h $(INSTALLINCDIR)
+	mkdir -p $(DESTDIR)$(INSTALLBINDIR)
+	mkdir -p $(DESTDIR)$(INSTALLLIBDIR)
+	mkdir -p $(DESTDIR)$(INSTALLINCDIR)
+	$(INSTALL) -m 644 include/*.h $(DESTDIR)$(INSTALLINCDIR)
 	@ $(MAKE) -C main --no-print-directory install
 
 time-stamp :
--- flite-1.4-release/main/Makefile.orig	2009-08-14 23:11:07.000000000 +0200
+++ flite-1.4-release/main/Makefile	2012-10-06 14:50:38.057184532 +0200
@@ -111,15 +111,15 @@
 
 install:
 #       The basic binaries
-	@ $(INSTALL) -m 755 $(BINDIR)/flite $(INSTALLBINDIR)
+	@ $(INSTALL) -m 755 $(BINDIR)/flite $(DESTDIR)$(INSTALLBINDIR)
 	@ for i in $(VOICES) ; \
 	do \
-	   $(INSTALL) $(BINDIR)/flite_$$i $(INSTALLBINDIR); \
+	   $(INSTALL) $(BINDIR)/flite_$$i $(DESTDIR)$(INSTALLBINDIR); \
 	done
-	$(INSTALL) -m 755 $(BINDIR)/flite_time $(INSTALLBINDIR)
+	$(INSTALL) -m 755 $(BINDIR)/flite_time $(DESTDIR)$(INSTALLBINDIR)
 #       The libraries: static and shared (if built)
-	cp -pd $(flite_LIBS_deps) $(INSTALLLIBDIR)
+	cp -pd $(flite_LIBS_deps) $(DESTDIR)$(INSTALLLIBDIR)
 ifdef SHFLAGS
-	cp -pd $(SHAREDLIBS) $(VERSIONSHAREDLIBS) $(INSTALLLIBDIR)
+	cp -pd $(SHAREDLIBS) $(VERSIONSHAREDLIBS) $(DESTDIR)$(INSTALLLIBDIR)
 endif
 
