diff -urNp -x '*.orig' flite-2.2.org/configure.in flite-2.2/configure.in
--- flite-2.2.org/configure.in	2020-08-13 02:17:09.000000000 +0200
+++ flite-2.2/configure.in	2024-12-07 13:22:19.026052513 +0100
@@ -423,8 +423,7 @@ AC_ARG_WITH( audio,
         AUDIODRIVER=$with_audio
 	AUDIODEFS="" )
 
-if test "x$AUDIODEFS" = x; then
-    case "$AUDIODRIVER" in
+case "$AUDIODRIVER" in
 	linux|oss)
 	    AUDIODRIVER=oss
 	    AUDIODEFS=-DCST_AUDIO_LINUX
@@ -450,8 +449,7 @@ if test "x$AUDIODEFS" = x; then
 	none)
 	    AUDIODEFS=-DCST_AUDIO_NONE
 	    ;;
-    esac
-fi
+esac
 AC_SUBST(AUDIODRIVER)
 AC_SUBST(AUDIODEFS)
 AC_SUBST(AUDIOLIBS)
